samples/kotlin-android-app/src/main/java/com/amplitude/android/sample/MainApplication.kt

```diff
@@ -0,0 +1,44 @@
+package com.amplitude.android.sample
+
+import android.app.Application
+import com.amplitude.android.Amplitude
+import com.amplitude.android.Configuration
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.platform.Plugin
+import org.json.JSONObject
+
+class MainApplication : Application() {
+    companion object {
+        lateinit var amplitude: Amplitude
+        const val AMPLITUDE_API_KEY = BuildConfig.AMPLITUDE_API_KEY
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        // init instance
+        amplitude = Amplitude(
+            Configuration(
+                apiKey = AMPLITUDE_API_KEY,
+                context = applicationContext
+            )
+        )
+
+        // add sample plugin
+        amplitude.add(object : Plugin {
+            override val type: Plugin.Type = Plugin.Type.Enrichment
+            override lateinit var amplitude: com.amplitude.core.Amplitude
```

Comment:
Is the `lateinit amplitude` override necessary?

Reply:
That is still needed, there is a `var amplitude` inside Plugin interface, and the amplitude instance is not filled until the setup call.